Source/Core/Core/IOS/DI/DI.cpp

```diff
@@ -30,93 +52,679 @@ void DI::DoState(PointerWrap& p)
+          end <= range.second)
+      {
+        DICMDBUF0 = 0xA8000000;
+        DICMDBUF1 = position;
+        DICMDBUF2 = length;
+        return StartDMATransfer(length, request);
+      }
+    }
+    WARN_LOG(IOS_DI, "DVDLowUnencryptedRead: trying to read from an illegal region!");
+    return DIResult::SecurityError;
+  }
+  case DIIoctl::DVDLowEnableDvdVideo:
+    ERROR_LOG(IOS_DI, "DVDLowEnableDvdVideo - rejecting");
+    return DIResult::SecurityError;
+  // There are a bunch of ioctlvs that are also (unintentionally) usable as ioctls; reject these in
+  // Dolphin as games are unlikely to use them.
+  case DIIoctl::DVDLowGetNoDiscOpenPartitionParams:
+    ERROR_LOG(IOS_DI, "DVDLowGetNoDiscOpenPartitionParams as an ioctl - rejecting");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    return DIResult::SecurityError;
+  case DIIoctl::DVDLowNoDiscOpenPartition:
+    ERROR_LOG(IOS_DI, "DVDLowNoDiscOpenPartition as an ioctl - rejecting");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    return DIResult::SecurityError;
+  case DIIoctl::DVDLowGetNoDiscBufferSizes:
+    ERROR_LOG(IOS_DI, "DVDLowGetNoDiscBufferSizes as an ioctl - rejecting");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    return DIResult::SecurityError;
+  case DIIoctl::DVDLowOpenPartitionWithTmdAndTicket:
+    ERROR_LOG(IOS_DI, "DVDLowOpenPartitionWithTmdAndTicket as an ioctl - rejecting");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    return DIResult::SecurityError;
+  case DIIoctl::DVDLowOpenPartitionWithTmdAndTicketView:
+    ERROR_LOG(IOS_DI, "DVDLowOpenPartitionWithTmdAndTicketView as an ioctl - rejecting");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    return DIResult::SecurityError;
+  case DIIoctl::DVDLowGetStatusRegister:
+  {
+    u32 disr = DISR;
+    INFO_LOG(IOS_DI, "DVDLowGetStatusRegister: 0x%08x", disr);
+    return WriteIfFits(request, disr);
+  }
+  case DIIoctl::DVDLowGetControlRegister:
+  {
+    u32 dicr = DICR;
+    INFO_LOG(IOS_DI, "DVDLowGetControlRegister: 0x%08x", dicr);
+    return WriteIfFits(request, dicr);
+  }
+  case DIIoctl::DVDLowReportKey:
+  {
+    const u8 param1 = Memory::Read_U8(request.buffer_in + 7);
+    const u32 param2 = Memory::Read_U32(request.buffer_in + 8);
+    INFO_LOG(IOS_DI, "DVDLowReportKey: param1 0x%02x, param2 0x%06x", param1, param2);
+    DICMDBUF0 = 0xA4000000 | (param1 << 16);
+    DICMDBUF1 = param2 & 0xFFFFFF;
+    DICMDBUF2 = 0;
+    return StartDMATransfer(0x20, request);
+  }
+  case DIIoctl::DVDLowSeek:
+  {
+    const u32 position = Memory::Read_U32(request.buffer_in + 4);  // 32-bit offset
+    INFO_LOG(IOS_DI, "DVDLowSeek: position 0x%08x, translated to 0x%08x", position,
+             position);  // TODO: do partition translation!
```

Comment:
I want to wait with it, since the actual seek command isn't implemented yet so there isn't much point in doing it yet.  The comment is mainly to note that it should be done here, or at least only done for Wii games and not Gamecube games.